Mathlib/Data/Vector/Basic.lean

```diff
@@ -511,9 +511,25 @@ def casesOn₃ {motive : ∀ {n}, Vector α n → Vector β n → Vector γ n 
     motive v₁ v₂ v₃ :=
   inductionOn₃ (C := motive) v₁ v₂ v₃ nil @fun _ x y z xs ys zs _ => cons x y z xs ys zs
 
-/-- Cast a vector to an array. -/
-def toArray : Vector α n → Array α
-  | ⟨xs, _⟩ => cast (by rfl) xs.toArray
+/-- Convert a `List.Vector` to a `Vector`. -/
+def toVector (v : List.Vector α n) : _root_.Vector α n := ⟨v.1.toArray, v.2⟩
```

Comment:
Does `@[simps]` generate anything useful? Ideally we want the lemmas `v,toVector.get = v.get`, `v.toVector[i] = v[i]`, `v.toVector.toArray = v.toList.toArray`, etc.

Reply:
It gives `toVector_toList` and `ofVector_coe`. Which are fine, but not all of those things. It would be good to have a definitive list of lemmas to capture. Obviously I can see the ones you suggest are true but I would prefer to have a sense of what is best and comprehensive. This could be added in a separate PR though.

(What would be quite nice is a notion of "a map/equivalence that preserves `get/getElem`" but that feels more involved.)

Reply:
The definitive list is: If you have a vector operation parallel to a list operation, then `vector.toList.op = vector.op` and `list.toVector.op = list.op` should be theorems.

Reply:
Have made some progress; some lemmas were annoying to prove (especially without importing a bunch more stuff which I felt I ought to avoid) so I didn't do that. But I have the `get` and `getElem` lemmas and more besides.